share/p2p/shrexnd/server.go

```diff
@@ -0,0 +1,184 @@
+package shrexnd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/minio/sha256-simd"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds"
+	"github.com/celestiaorg/celestia-node/share/getters"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+	pb "github.com/celestiaorg/celestia-node/share/p2p/shrexnd/pb"
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+)
+
+const ndProtocolID = "/shrex/nd/0.0.1"
+
+var log = logging.Logger("shrex/nd")
+
+const (
+	writeTimeout = time.Second * 10
+	readTimeout  = time.Second * 10
+	serveTimeout = time.Second * 10
+)
```

Comment:
Can we make this to be params like we did for shrex/eds? It does not have to be registered in our nodebuilder config, but it still should be configurable.

Reply:
This will also eliminate the requirement to store them on the server and the serve can just hold params structure